spec/models/state_changer/standard_plate_spec.rb

```diff
@@ -84,5 +84,138 @@
         expect(reloaded_library_requests).to all(be_customer_accepts_responsibility)
       end
     end
+
+    context 'when a specific well is failed' do
+      include_context 'a limber target plate with submissions'
+      let(:target_state) { 'failed' }
+      let(:contents) { ['A1'] } # This is the well that will be failed
+
+      it 'allows failure of the specified requests', :aggregate_failures do
+        expect { state_changer.update_labware_state }.not_to change(BroadcastEvent::LibraryStart, :count)
+        reloaded_library_requests = library_requests.each(&:reload)
+
+        # A1 request should be failed
+        expect(reloaded_library_requests.first).to be_failed
+
+        # Other requests in B1 and C1 should be unchanged
+        expect([reloaded_library_requests[1], reloaded_library_requests.last]).to all(be_started)
+      end
+    end
+
+    context 'when a specific well is failed but the request is shared by other wells' do
```

Comment:
Small refactoring suggestions, saves a few lines. Only functionality change is that when you're grabbing the parent well requests, I think you should be specifying the request type (library), as limber_shared_context seems to add multiplexing requests too - I've done that below using `library_requests`.
```
include_context 'a limber target plate with submissions'
      let(:tested_wells) { 2 }
      let(:target_state) { 'failed' }

      # override default target plate creation
      let(:target_plate) { create :plate, well_count: 4 }
      let(:target_wells) { target_plate.wells.index_by(&:map_description) }

      before do
        parent_wells = input_plate.wells.index_by(&:map_description)

        source_well_a1_request = library_requests.detect { |r| r.source_well == parent_wells['A1'] } # or just library_requests[0]
        source_well_b1_request = library_requests.detect { |r| r.source_well == parent_wells['B1'] } # or just library_requests[1]

        # set up wells in target plate
        # A1, B1 and C1 created by transferring from parent A1 (1st library request)
        target_wells['A1'].stock_well_links <<
          build(:stock_well_link, target_well: target_wells['A1'], source_well: parent_wells['A1'])
        create :transfer_request,
               asset: parent_wells['A1'],
               target_asset: target_wells['A1'],
               outer_request: source_well_a1_request

        target_wells['B1'].stock_well_links <<
          build(:stock_well_link, target_well: target_wells['B1'], source_well: parent_wells['A1'])
        create :transfer_request,
               asset: parent_wells['A1'],
               target_asset: target_wells['B1'],
               outer_request: source_well_a1_request

        target_wells['C1'].stock_well_links <<
          build(:stock_well_link, target_well: target_wells['C1'], source_well: parent_wells['A1'])
        create :transfer_request,
               asset: parent_wells['A1'],
               target_asset: target_wells['C1'],
               outer_request: source_well_a1_request

        # D1 created by transferring from parent B1 (2nd library request)
        target_wells['D1'].stock_well_links <<
          build(:stock_well_link, target_well: target_wells['D1'], source_well: parent_wells['B1'])
        create :transfer_request,
               asset: parent_wells['B1'],
               target_asset: target_wells['D1'],
               outer_request: source_well_b1_request
      end
```

Reply:
Thanks, changed.
Had originally created a parent plate with only a library request, not multiplexing, but couldn't get it all to fully work (factory pain).